### PR TITLE
Nat/unified

### PR DESCRIPTION
**Pull request by nturinski**

### CODE REVIEW

**Review by alexweininger — APPROVED** [review]
You don't have to do it in this PR, but we added the api.d.ts typings to the utils package https://github.com/microsoft/vscode-azuretools/blob/main/utils/hostapi.d.ts

Some stuff was renamed, the old names are marked as deprecated.

**Comment by nturinski** [comment]
Got it!

Also left to do: contribute a local provider for the attached database accounts/emulator.  I'm not sure whether or not "attached" (that is, a remote database but attached via connection string) should go into the Azure or Workspace view though.

File: src/resolver/ResolvedDatabaseAccountResource.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import { AzExtTreeItem, IActionContext, ICreateChildImplContext, TreeItemIconPath } from "@microsoft/vscode-azext-utils";
+import { AppResource, ResolvedAppResourceBase } from "../api";
+import { DocDBAccountTreeItemBase } from "../docdb/tree/DocDBAccountTreeItemBase";
+import { MongoAccountTreeItem } from "../mongo/tree/MongoAccountTreeItem";
+import { PostgresServerTreeItem } from "../postgres/tree/PostgresServerTreeItem";
+
+export class ResolvedDatabaseAccountResource implements ResolvedAppResourceBase {
+    public id: string;
+    public contextValuesToAdd?: string[] | undefined = [];
```
Comment (bwateratmsft) [inline src/resolver/ResolvedDatabaseAccountResource.ts:14]: If we're initializing here then it's guaranteed to have a value; we don't need the `?` / `| undefined`, and the `?`s later on.

**Review by bwateratmsft — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -90,15 +90,27 @@
...
         "onLanguage:mongo",
-        "onView:azureDatabasesExplorer"
+        "onView:azureResourceGroups"
     ],
     "contributes": {
-        "views": {
-            "azure": [
+        "x-azResources": {
+            "activation": {
+                "onFetch": [
```
Comment (bwateratmsft) [inline package.json:98]: Just `onResolve` is sufficient unless the activation needs a head start.

**Comment by nturinski** [comment]
Will do Alex's feedback in another PR

**Pull request merged**